python/tank/util/local_file_storage.py

```diff
@@ -84,7 +84,9 @@ def get_global_root(cls, path_type, generation=CORE_V18):
                            which is the current generation of paths.
         :return: Path as string
         """
-        if generation == cls.CORE_V18:
+        # If SHOTGUN_HOME is set then the intent is to not use any of official locations and instead
+        # use an alternate location, in which case this is as if we're running core 0.18.
+        if generation == cls.CORE_V18 or os.environ.get("SHOTGUN_HOME"):
```

Comment:
Consider the session cache. It wants to read authentication tokens from the legacy session cache and then the new session cache location. When you set SHOTGUN_HOME, the code will then read inside the legacy session cache and then under SHOTGUN_HOME.

I would argue that when someone sets SHOTGUN_HOME, they are not interested into the legacy behaviour. They want everything to work in that sandbox. It means credentials, caches and everything should be read from that location. They expect everything to we self-contained at that location and not leak into it through legacy code paths.

At leasts, that's my assumption. Do you feel that is a reasonable one? I know for sure that when our unit tests run we certainly don't want things to leak out of SHOTGUN_HOME, otherwise we end up with the tests modifying out authentication.yml files and what not.